### PR TITLE
Fix #61 - Remove double output when cljx auto task is executed.

### LINKED ISSUE #61: cljx auto generates double output
I'm using cljx 0.5.0 and it works great when I run `lein cljx once`. However, when I run `lein cljx auto` it will generate the code twice and I'm not sure wether this is caused by a misconfiguration in the `project.clj` or a bug of cljx.

The following folders are created:

1. `target/generated-src/{ns}...` -> correct clj/cljs files
2. `target/generated-src/file//{path-to-project}/test/cljx/{ns}/...` -> contains generated clj/cljs test-files
3. `target/generated-test/{ns}...` -> correct clj/cljs test-files
4. `target/generated-src/file//{path-to-project}/src/cljx/{ns}/...` -> contains generated clj/cljs files

With `lein cljx once` only the folders of 1 and 3 are generated.

My project.clj contains the following:

```
(defproject ...
  :source-paths ["src/clj" "src/cljs" "target/generated-src"]
  :test-paths ["test/clj" "test/cljs" "target/generated-test"]
...
  :auto-clean false
  :cljx {:builds [{:source-paths ["src/cljx"]
                   :output-path "target/generated-src"
                   :rules :clj}
                  {:source-paths ["src/cljx"]
                   :output-path "target/generated-src"
                   :rules :cljs}
                  {:source-paths ["test/cljx"]
                   :output-path "target/generated-test"
                   :rules :clj}
                  {:source-paths ["test/cljx"]
                   :output-path "target/generated-test"
                   :rules :cljs}]}
...
  :dependencies [
    [org.clojure/clojure            "1.6.0"]
    [org.clojure/clojurescript      "0.0-2511" :scope "provided"]
...
  ]
  :profiles {
    :dev {
      :plugins [[com.keminglabs/cljx  "0.5.0"          :exclusions [org.clojure/clojure org.clojure/clojurescript]]]
      :prep-tasks   [["cljx" "once"]]
...
  }}
}})

### PR DESCRIPTION
**Pull request by niwibe**
The bug is mainly caused by wrong relativize behavior when a file
does not corresponds to the same rootpath, appending unexpected
and/or wrong handled `file://` prefix.

This fix, adds additional step that filters the incoming file list
by corresponding rootpath, allowing to relativize working correctly
and add additionally strip the `file://` prefix from result of
relativize if it appears unexpectly.

NOTE: replaces #62, now compatible with jdk6 and in one unique commit.

### CODE REVIEW

**Comment by Deraen** [comment]
:+1: 

This also fixes using relative source-paths, like "../shared/src/cljx".

File: src/cljx/core.clj [hunk 1]
```diff
@@ -42,41 +45,62 @@ Returns a sequence of File objects, in breadth-first sort order."
...
+(defn- filter-files-for-path
+  "Filter a bunch of files that matches the
+  provided path."
+  [files sourcepath]
+  (let [^URI sourceuri (.toURI (io/file sourcepath))
+        ^String spath (util/path sourceuri)]
+    (filter (fn [f]
+              (let [^URI uri (.toURI f)
+                    ^String fpath (util/path uri)]
+                (.startsWith fpath sourcepath)))
```
Comment (Deraen) [inline src/cljx/core.clj]: Should sourcepath be spath instead?

Comment (niwibe) [inline src/cljx/core.clj]: Maybe it can be changed to spath, but is a minor issue. We can fix it in other PRs.

Comment (Deraen) [inline src/cljx/core.clj]: Looks like major issue to me. `fpath` is absolute path to the file and `sourcepath` is relative path to source directory, thus no files are matched.

Could you check if you can reproduce my problem e.g. by removing the output directory, running `lein cljx once` and checking if any files were written.

Comment (niwibe) [inline src/cljx/core.clj]: You are completely right. I thought that you says about spelling only... my mistake!

Comment (niwibe) [inline src/cljx/core.clj]: Fixed

**Comment by cemerick** [comment]
Thank you for your work in this @niwibe, but the proposed changes were quite large for the problem at hand. I believe I've fixed this on master, available in `[org.clojars.cemerick/cljx "0.6.0-SNAPSHOT"]` until the final 0.6.0 release is cut.

@Deraen The change I just made does seem to work well with relative source paths like you suggest…however, I wouldn't rely on such things to work. Projects should be contained within a particular directory; I'd suggest using symlinks to bring in other directories if you absolutely have to, rather than using relative paths.

**Pull request closed without merging**